src/main/java/com/google/cloud/resourcemanager/Lien.java

```diff
@@ -0,0 +1,183 @@
+    public Builder setCreateTime(String createTime) {
+      infoBuilder.setCreateTime(createTime);
+      return this;
+    }
+
+    @Override
+    public Builder setName(String name) {
+      infoBuilder.setName(name);
+      return this;
+    }
+
+    @Override
+    public Builder setOrigin(String origin) {
+      infoBuilder.setOrigin(origin);
+      return this;
+    }
+
+    @Override
+    public Builder setParent(String parent) {
+      infoBuilder.setParent(parent);
+      return this;
+    }
+
+    @Override
+    public Builder setReason(String reason) {
+      infoBuilder.setReason(reason);
+      return this;
+    }
+
+    @Override
+    public Builder setRestrictions(List<String> restrictions) {
+      infoBuilder.setRestrictions(restrictions);
+      return this;
+    }
+
+    @Override
+    public Lien build() {
+      return new Lien(resourceManager, infoBuilder);
+    }
+  }
+
+  Lien(ResourceManager resourceManager, LienInfo.BuilderImpl infoBuilder) {
+    super(infoBuilder);
+    this.resourceManager = checkNotNull(resourceManager);
+    this.options = resourceManager.getOptions();
+  }
+
+  /** Returns the {@link ResourceManager} service object associated with this Lien. */
+  public ResourceManager getResourceManager() {
+    return resourceManager;
+  }
+
+  /**
+   * Creates the Lien which applies to the resource denoted by the parent field.
+   *
+   * <p>Callers of this method require permission on the parent resource. For example, applying to
+   * projects/1234 requires permission resourcemanager.projects.updateLiens.
+   *
+   * <p>Some resources may limit the number of Liens which may be applied.
+   *
+   * @throws ResourceManagerException upon failure
+   * @see <a href= "https://cloud.google.com/resource-manager/reference/rest/v1/liens/create">Cloud
+   *     Resource Manager create</a>
+   */
+  public Lien create(LienInfo lienInfo) {
+    return resourceManager.createLien(lienInfo);
+  }
+
+  /**
+   * Deletes the lien by name.
+   *
+   * <p>Callers of this method require permission on the parent resource. For example, a Lien with a
+   * parent of projects/1234 requires permission resourcemanager.projects.updateLiens.
+   *
+   * @throws ResourceManagerException upon failure
+   * @see <a href= "https://cloud.google.com/resource-manager/reference/rest/v1/liens/delete">Cloud
+   *     Resource Manager delete</a>
+   */
+  public void delete() {
+    resourceManager.deleteLien(getName());
+  }
+
+  /**
+   * Retrieves the lien by name.
+   *
+   * <p>Callers of this method require permission on the parent resource. For example, a Lien with a
+   * parent of projects/1234 requires permission requires permission resourcemanager.projects.get or
+   * resourcemanager.projects.updateLiens.
+   *
+   * @throws ResourceManagerException upon failure
+   * @see <a href= "https://cloud.google.com/resource-manager/reference/rest/v1/liens/list">Cloud
+   *     Resource Manager get</a>
+   */
+  public Lien get() {
```

Comment:
This is really confusing. Since this returns a completely new object, why is this an instance method at all? Using the resourceManager directly seems far more appropriate. 

Also I'm 90% sure ResourceManagerException should be checked because it indicates an unpredictable, environmental failure.

Reply:
Done.thanks for this